enterprise/suppress-user/factory.go

```diff
@@ -26,57 +31,197 @@ func (m *Factory) Setup(ctx context.Context, backendConfig backendconfig.Backend
 	if m.Log == nil {
 		m.Log = logger.NewLogger().Child("enterprise").Child("suppress-user")
 	}
+
 	if m.EnterpriseToken == "" {
 		m.Log.Info("Suppress User feature is enterprise only")
 		return &NOOP{}, nil
 	}
+
 	m.Log.Info("Setting up Suppress User Feature")
+
 	backendConfig.WaitForConfig(ctx)
-	var repository Repository
-	if config.GetBool("BackendConfig.Regulations.useBadgerDB", true) {
-		tmpDir, err := misc.CreateTMPDIR()
+
+	var pollInterval time.Duration
+	config.RegisterDurationConfigVariable(300, &pollInterval, true, time.Second, "BackendConfig.Regulations.pollInterval")
+
+	useBadgerDB := config.GetBool("BackendConfig.Regulations.useBadgerDB", true)
+	if useBadgerDB {
+		identity := backendConfig.Identity()
+
+		fullSuppressionPath, latestSuppressionPath, err := getRepoPath()
 		if err != nil {
-			return nil, fmt.Errorf("could not create tmp dir: %w", err)
+			return nil, fmt.Errorf("could not get repo path: %w", err)
 		}
-		path := path.Join(tmpDir, "suppression")
 
-		// TODO: implement seeder source, to retrieve the initial state from a persisted backup
-		var seederSource func() (io.Reader, error)
+		if !alreadySeeded(fullSuppressionPath) && config.IsSet("SUPPRESS_USER_BACKUP_SERVICE_URL") {
+			repo := &RepoSwitcher{
+				mu: sync.RWMutex{},
+			}
+			// First starting a repository seeded with the latest data which is faster to load
+			latestSyncer, latestRepo, err := m.newSyncerWithBadgerRepo(latestSuppressionPath, latestDataSeed, config.GetDuration("BackendConfig.Regulations.maxSeedWait", 5, time.Second), identity, pollInterval)
+			if err != nil {
+				return nil, err
+			}
 
-		repository, err = NewBadgerRepository(
-			path,
-			m.Log,
-			WithSeederSource(seederSource),
-			WithMaxSeedWait(config.GetDuration("BackendConfig.Regulations.maxSeedWait", 5, time.Second)))
-		if err != nil {
-			return nil, fmt.Errorf("could not create badger repository: %w", err)
+			subCtx, latestSyncCancel := context.WithCancel(ctx)
+			rruntime.Go(func() {
+				latestSyncer.SyncLoop(subCtx)
+				err = latestRepo.Stop()
+				if err != nil {
+					m.Log.Error("Latest Sync failed: could not stop repo: %w", err)
+				}
+				err = os.RemoveAll(latestSuppressionPath)
+				if err != nil {
+					m.Log.Error("Latest Sync failed: could not remove repo: %w", err)
+				}
+			})
+			repo.Repository = latestRepo
+			rruntime.Go(func() {
+				var fullSyncer *Syncer
+				var fullRepo Repository
+				var err error
+
+				m.retry(ctx,
+					func() error {
+						fullSyncer, fullRepo, err = m.newSyncerWithBadgerRepo(fullSuppressionPath, fullDataSeed, 0, identity, pollInterval)
+						return err
+					}, 5*time.Second)
+
+				m.retry(ctx,
+					func() error { return fullSyncer.Sync(ctx) },
+					5*time.Second)
+				_, err = os.Create(filepath.Join(fullSuppressionPath, model.SyncDoneMarker))
+				if err != nil {
+					m.Log.Error("Could not create sync done marker: %w", err)
```

Comment:
well `sync` marker is just to indicate that syncing is done. So, even if we fail to create it and simply log the error, logically nothing would be wrong. Since, the sync actually completed successfully. The only issue would be that the next time when the server starts after a crash it won't be able to find the marker and hence would delete this directory and would try syncing again from the backup service which would be redundant.

I will try to put in int `retryIndefinitely` to solve this.